core/user/user.go

```diff
@@ -33,6 +32,6 @@ func (u *User) Validate() error {
 type Repository interface {
 	Create(ctx context.Context, u *User) (string, error)
 	GetByEmail(ctx context.Context, email string) (User, error)
-	GetByUUID(ctx context.Context, uuid string) (User, error)
-	UpsertByEmail(ctx context.Context, u *User) (string, error)
+	InsertByEmail(ctx context.Context, u *User) (string, error)
+	GetOrInsertByEmail(ctx context.Context, u *User) (string, error)
```

Comment:
what's the differences with `UpsertByEmail` ?

Reply:
There is no update query in `GetOrInsertByEmail`, only. get the userID if email already exist OR insert a new row if email not found in DB